BUG: quoted-strings only-when-needed with hashtag

Setup
---

_test.yml_:
```yaml
---
hashtag: 'Test #hashtag'
```

_.yamllint.yml_:  
```yaml
extends: default

rules:
  quoted-strings:
    required: only-when-needed
```

Result
---

```
$ yamllint test.yml 
test.yml
  2:10      error    string value is redundantly quoted with any quotes  (quoted-strings)
```

I don't know what the YAML spec says about this, but when I test using such value, it *is* parsed as an octal value.

Also `yamllint` reports it as a forbidden implicit octal value:
```
  1:16      error    forbidden implicit octal value "0123456789"  (octal-values)
```

--- a/tests/rules/test_quoted_strings.py
+++ b/tests/rules/test_quoted_strings.py
@@ -330,7 +330,8 @@ class QuotedTestCase(RuleTestCase):
                    '- "%wheel ALL=(ALL) NOPASSWD: ALL"\n'
                    '- \'"quoted"\'\n'
                    '- "\'foo\' == \'bar\'"\n'
-                   '- "\'Mac\' in ansible_facts.product_name"\n',
+                   '- "\'Mac\' in ansible_facts.product_name"\n'
+                   '- \'foo # bar\'\n',
                    conf)
         self.check('---\n'
                    'k1: ""\n'

--- a/yamllint/rules/quoted_strings.py
+++ b/yamllint/rules/quoted_strings.py
@@ -160,7 +160,7 @@ def _quotes_are_needed(string):
     try:
         a, b = loader.get_token(), loader.get_token()
         if (isinstance(a, yaml.ScalarToken) and a.style is None and
-                isinstance(b, yaml.BlockEndToken)):
+                isinstance(b, yaml.BlockEndToken) and a.value == string):
             return False
         return True
     except yaml.scanner.ScannerError:

